core/primitives/src/receipt.rs

```diff
@@ -141,15 +141,15 @@ const STATE_STORED_RECEIPT_TAG: u8 = u8::MAX;
 /// into the state. In that case the receipt should never need to be cloned. The
 /// serialization only needs a reference.
 #[derive(PartialEq, Eq, Debug, ProtocolSchema)]
-pub enum ReceiptOrStateStoredReceipt<'a> {
-    Receipt(Cow<'a, Receipt>),
-    StateStoredReceipt(StateStoredReceipt<'a>),
+pub enum ReceiptOrStateStoredReceipt {
+    Receipt(Receipt),
+    StateStoredReceipt(StateStoredReceipt),
 }
```

Comment:
The protocol schema CI check is failing because structs with `ProtocolSchema` were updated without updating the checksum file. This is a sanity check to ensure that the serialization format of structs that are part of the protocol doesn't accidentally change. Please ensure that modifying the struct doesn't change the serialization format (you can borsh serialize it before and after the change and compare) and update the checksums.

See: https://github.com/near/nearcore/tree/master/tools/protocol-schema-check

Reply:
nice to know, thanks!

Reply:
could confirm that removing the Cows doesn't change the data layout

Reply:
The protocol schema check is great. But is it possible that there could be a false false-positive (author decides a change of schema doesn't alter the data layout or it's backward compatible, but they are wrong)? When the checks fail, it's still up to developers to decide whether migration is needed. Cows here are alright, the serialization is straightforward, kinda obvious it wouldn't change the serialized format and it's easy to verify. But for more intricate cases, is there a safety net to ensure the data layout actually doesn't change? I may be overthinking. :))

Reply:
It's possible but in my experience it's more likely that a developer forgets to add ProtocolSchema to a new structure. 

We have canary nodes running on master (updated nightly) so if there is any protocol upgrade that isn't correctly handled we would see that. Then we have pre-release tests where we spin up a small network and gradually update nodes. Finally there is testnet and we really shouldn't break anything there but it is the final line of defense before mainnet.